src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -64,38 +72,41 @@ public int getNumberOfKeyOccurrences(String key) {
 
     }
 
-    //========================================================
-    // keep track of all the keys to warn if there are duplicates
-    //========================================================
-    public boolean addKeyToSet(String key) {
+    /**
+     * Keep track of all the keys to warn if there are duplicates
+     *
+     * @param key as String
+     * @return true, if the key already exists in the allkeys set, else false
+     */
+    protected boolean addKeyToSet(String key) {
         if ((key == null) || key.isEmpty()) {
             return false;//don't put empty key
         }
         boolean exists = false;
         if (allKeys.containsKey(key)) {
             // warning
             exists = true;
-            allKeys.put(key, allKeys.get(key) + 1);// incrementInteger( allKeys.get(key)));
+            allKeys.put(key, allKeys.get(key) + 1);
         } else {
             allKeys.put(key, 1);
         }
         return exists;
     }
 
-    //========================================================
-    // reduce the number of keys by 1. if this number goes to zero then remove from the set
-    // note: there is a good reason why we should not use a hashset but use hashmap instead
-    //========================================================
-    public void removeKeyFromSet(String key) {
+    /**
+     * Reduce the number of keys by 1. if this number goes to zero then remove from the set
+     * note: There is a good reason why we should not use a hashset but use hashmap instead
```

Comment:
What is the purpose of this note? The hashmap isn't even declared in this methode.